protocol/messenger.go

```diff
@@ -917,9 +924,68 @@ func (m *Messenger) Start() (*MessengerResponse, error) {
 		}
 	}
 
+	if m.processBackedupMessages {
+		err = m.startBackupFetchingTracking(response)
+		if err != nil {
+			return nil, err
+		}
+	}
+
 	return response, nil
 }
 
+func (m *Messenger) startBackupFetchingTracking(response *MessengerResponse) error {
+	// Add an acivity center notification to show that we are fetching back up messages
+	notification := &ActivityCenterNotification{
+		ID:        types.FromHex(backupSyncingNotificationID), // TODO put this in a constant
+		Type:      ActivityCenterNotificationTypeBackupSyncingFetching,
+		Timestamp: m.getTimesource().GetCurrentTime(),
+		Read:      false,
+		Deleted:   false,
+		UpdatedAt: m.GetCurrentTimeInMillis(),
+	}
+	err := m.addActivityCenterNotification(response, notification, nil)
+
+	if err != nil {
+		return err
+	}
+
+	// Add a timeout to mark the backup syncing as failed after 1 minute and 30 seconds
+	time.AfterFunc(1*time.Minute+30*time.Second, func() {
+		if m.fetchingBackedUpDataCompleted {
```

Comment:
Doesn't it need a mutex or the risk of race condition is neglectable?

Reply:
Good point, will do